File: test-fixtures/languages/java/complex/DecoratedDeepFlowEntry.java
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.util.Map;

@Retention(RetentionPolicy.RUNTIME)
@interface DfcWrappedAudited {}

public class DecoratedDeepFlowEntry {
    @DfcWrappedAudited
    public static String runDecorated(String rawInput) {
        DeepFlowChain.DfcCommandParser parser = new DeepFlowChain.DfcCommandParser();
        DeepFlowChain.DfcInputValidator validator = new DeepFlowChain.DfcInputValidator();

        Map<String, String> tokens = parser.tokenize(rawInput);
        Map<String, String> descriptor = parser.normalize(tokens);
        Map<String, String> validated = validator.checkFormat(descriptor);
        DfcRequestContext ctx = DeepFlowChain.dfcBuildRequest(validated);
        DfcRequestContext processed = DeepFlowChain.dfcApplyMiddleware(ctx);
        return DeepFlowChain.dfcDispatchToExecutor(processed);
    }
}

File: test-fixtures/languages/typescript/complex/decorated_deep_flow_entry.ts
import { dfcMain } from "./deep_flow_chain";

function Audited(
  _target: object,
  _propertyKey: string,
  descriptor: PropertyDescriptor,
): PropertyDescriptor {
  return descriptor;
}

export class DfcDecoratedEntry {
  @Audited
  async runDecorated(): Promise<void> {
    await dfcMain();
  }
}
